dbgpt/agent/agents/expand/retrieve_summary_assistant_agent.py

```diff
@@ -0,0 +1,570 @@
+                if "NO RELATIONSHIP." in message:
+                    fail_reason = f"Return summarization error, the provided text content has no relationship to user's question. TERMINATE."
+                    response_success = False
+                else:
+                    content = message
+                    view = content
+            except Exception as e:
+                fail_reason = f"Return summarization error, {str(e)}"
+                response_success = False
+
+        if not response_success:
+            content = fail_reason
+        return True, {
+            "is_exe_success": response_success,
+            "content": content,
+            "view": view,
+        }
+
+    def _get_files_from_dir(
+        self,
+        dir_path: Union[str, List[str]],
+        types: list = TEXT_FORMATS,
+        recursive: bool = True,
+    ):
+        """Return a list of all the files in a given directory, a url, a file path or a list of them."""
+        if len(types) == 0:
+            raise ValueError("types cannot be empty.")
+        types = [t[1:].lower() if t.startswith(".") else t.lower() for t in set(types)]
+        types += [t.upper() for t in types]
+
+        files = []
+        # If the path is a list of files or urls, process and return them
+        if isinstance(dir_path, list):
+            for item in dir_path:
+                if os.path.isfile(item):
+                    files.append(item)
+                elif self._is_url(item):
+                    files.append(self._get_file_from_url(item))
+                elif os.path.exists(item):
+                    try:
+                        files.extend(self._get_files_from_dir(item, types, recursive))
+                    except ValueError:
+                        logger.warning(f"Directory {item} does not exist. Skipping.")
+                else:
+                    logger.warning(f"File {item} does not exist. Skipping.")
+            return files
+
+        # If the path is a file, return it
+        if os.path.isfile(dir_path):
+            return [dir_path]
+
+        # If the path is a url, download it and return the downloaded file
+        if self._is_url(dir_path):
+            return [self._get_file_from_url(dir_path)]
+
+        if os.path.exists(dir_path):
+            for type in types:
+                if recursive:
+                    files += glob.glob(
+                        os.path.join(dir_path, f"**/*.{type}"), recursive=True
+                    )
+                else:
+                    files += glob.glob(
+                        os.path.join(dir_path, f"*.{type}"), recursive=False
+                    )
+        else:
+            logger.error(f"Directory {dir_path} does not exist.")
+            raise ValueError(f"Directory {dir_path} does not exist.")
+        return files
+
+    def _get_file_from_url(self, url: str, save_path: str = None):
+        """Download a file from a URL."""
+        current_directory = os.getcwd()
```

Comment:
We may need to put the temporary output of agents somewhere, maybe consider `pilot/data/agent_outputs`.

Reply:
Thanks for the suggestion!
@fangfcg has already changed the path to `PILOT_PATH`.
He also added the `a_verify` function which can be tested now. @csunny @fangyinc @yhjun1026